src/serve/javaEmulators.js

```diff
@@ -88,7 +89,8 @@ function _start(targetName) {
     utils.logWarning("Setup required, please run: firebase setup:emulators:" + emulator.name);
     return Promise.reject("emulator not found");
   }
-  return _runBinary(emulator, command);
+
+  return track("emulators:start", targetName).then(() => _runBinary(emulator, command));
```

Comment:
I wouldn't wait on the returned promise from `track()` before starting the emulator, as that will add latency to emulator startup. I'd just "fire and forget it" right before `_runBinary` instead.

Reply:
+1